### PR TITLE
Working pipeline with CLI

### PR DESCRIPTION
**Pull request by JacobPolloreno**
The pipeline and the CLI(main.py) work well for all stages but the trained model needs to be debugged. I'm also going to extract a lot of logic in main.py to make it more generalizable to a REST API and CLI.

There are no pressing issues with the code. I am currently working on more test cases and documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,106 +1,71 @@
...
-## Build Environment
-- Include instructions of how to launch scripts in the build subfolder
-- Build scripts can include shell scripts or python setup.py files
-- The purpose of these scripts is to build a standalone environment, for running the code in this repository
-- The environment can be for local use, or for use in a cloud environment
-- If using for a cloud environment, commands could include CLI tools from a cloud provider (i.e. gsutil from Google Cloud Platform)
-```
-# Example
+##### Dependencies
+- Python 3.6
```
Comment (mrubash1) [inline README.md]: do you need all of these dependcies before you run your bash build script? You can make your build script build a virtualenv that then downloads keras, pandas, click and pytest

File: build/run_build.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+echo "Office Answers Project"
+echo "----------------------"
+
+py=python3
+py36_error_msg="Python 3.6 is not installed.\nPlease install or modify this script to use proper Python executable"
+
+if command -v $py &>/dev/null; then
```
Comment (mrubash1) [inline build/run_build.sh:9]: you could pass this step by just using conda for managing the virtualenv in the future

but no need to change it :)

File: README.md [hunk 3]
```diff
@@ -1,106 +1,71 @@
...
 
-## Test
-- Include instructions for how to run all tests after the software is installed
-```
-# Example
-
-# Step 1
-# Step 2
-```
+#### Step 1: Configuration
```
Comment (mrubash1) [inline README.md:32]: Cool, thanks for having this!

File: data/raw/prepare_wikiqa.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    outfile = os.path.join(dstdir, 'raw_wiki_data.txt')
+
+    try:
+        with open(outfile, 'w') as f:
+            for infile in infiles:
+                with open(infile, 'r') as f2:
+                    for line in f2:
+                        r = line.strip().split('\t')
+                        f.write('%s\t%s\t%s\n' % (r[0], r[1], r[2]))
+    except FileExistsError as e:
```
Comment (mrubash1) [inline data/raw/prepare_wikiqa.py]: good work actually doing the try/except !

File: src/officeanswers/main.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+
+        click.echo('Saving corpus questions and documents...')
+        corpus_d_path = os.path.join(data_dir, 'documents.dill')
+        dill.dump(corpus_d, open(corpus_d_path, 'wb'))
+        corpus_q_path = os.path.join(data_dir, 'questions.dill')
+        dill.dump(corpus_d, open(corpus_q_path, 'wb'))
+
+    except KeyError:
+        logger.error(
+            "Config file doesn't have corpus path in inputs.share.corpus")
```
Comment (mrubash1) [inline src/officeanswers/main.py]: I don't think this error message is correct, as can there not be lots of different types of key errors? I would print out the specific key that is missing here if thats the case

File: tests/cli/test_cli.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import click
```
Comment (mrubash1) [inline tests/cli/test_cli.py]: What does this test actually do? I would just make it functional, to test that you can run the code for training or inference

**Comment by mrubash1** [comment]
Looks really good so far. I made a few comments, and I think 1 thing to fix in the readme, a request for another test, and a clarification question.

After those things satisfied, +1

File: README.md (hunk 1 above)
Comment (JacobPolloreno) [reply README.md]: I worked on that yesterday and have build scripts that I'll add to the commit

**Comment by JacobPolloreno** [comment]
I updated the readme to reflect the changes you suggested, removed the CLI test, and added a test for the whole pipeline.

**Pull request merged**